src/main/java/land/oras/credentials/FileStore.java

```diff
@@ -0,0 +1,166 @@
+package land.oras.credentials;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+import com.google.gson.reflect.TypeToken;
+import land.oras.exception.ConfigLoadingException;
+import land.oras.utils.JsonUtils;
+
+/**
+ * FileStore implements a credentials store using a configuration file
+ * to keep the credentials in plain-text.
+ *
+ * Reference: https://docs.docker.com/engine/reference/commandline/cli/#docker-cli-configuration-file-configjson-properties
+ */
+public class FileStore {
+
+    private final boolean disablePut;
+    private final Config config;
+
+    public static final String ERR_PLAINTEXT_PUT_DISABLED = "Putting plaintext credentials is disabled";
+    public static final String ERR_BAD_CREDENTIAL_FORMAT = "Bad credential format";
+
+    /**
+     * Constructor for FileStore.
+     *
+     * @param disablePut boolean flag to disable putting credentials in plaintext.
+     * @param config configuration instance.
+     */
+    public FileStore(boolean disablePut, Config config) {
```

Comment:
We have a static method to contruct the filestore `newFileStore`.

Can you explain me the reason why the contructor is public?

Reply:
I was creating some test case that's why i made this constructor public also can we allow the user to encrypt the credentials before storing into the files or directly allow them to store in the plain text